agent/consul/controller/controller.go

```diff
@@ -347,6 +349,13 @@ func (c *controller) enqueueEntry(entry structs.ConfigEntry, transformers ...Tra
 	}
 }
 
+// Enqueue adds all of the given entry into the work queue.
+func (c *controller) Enqueue(requests ...Request) {
```

Comment:
This allows us to be super flexible with the way we handle triggering reconciliation on dependent resources. It's used in the inline-certificate reconciliation code path, allowing us to retrieve all gateways that reference the given cert and then enqueue them for reconciliation.